frontend/packages/pipelines-plugin/src/components/repository/types.ts

```diff
@@ -0,0 +1,22 @@
+import { K8sResourceKind } from '@console/internal/module/k8s';
+import { Condition } from '../../types';
+
+export type RepositoryStatus = {
+  completionTime?: string;
+  conditions?: Condition[];
+  logurl?: string;
+  pipelineRunName: string;
+  sha?: string;
+  startTime?: string;
+  title?: string;
+};
+
+export type RepositoryKind = K8sResourceKind & {
+  spec: {
+    url: string;
+    branch?: string;
+    event_type?: string;
+    namespace?: string;
+  };
+  pipelinerun_status: RepositoryStatus[];
```

Comment:
Initially when you create `Repository` CR status field will not be available, make it optional.
```suggestion
  pipelinerun_status?: RepositoryStatus[];
```

Reply:
made it as optional